### PR TITLE
docs(style): update outdated `stylo` project name and link

### LINKED ISSUE #32: Dead links in style.md
https://github.com/servo/servo/blob/master/docs/components/style.md

These links no longer work:

http://doc.servo.org/style/selector_parser/trait.SelectorImplExt.html
http://doc.servo.org/style/servo_selector_parser/struct.SelectorImpl.html
http://doc.servo.org/selectors/matching/struct.Rule.html
http://doc.servo.org/selectors/matching/struct.SelectorMap.html
http://doc.servo.org/style/stylist/struct.PerPseudoElementSelectorMap.html
http://doc.servo.org/style/stylist/struct.PerOriginSelectorMap.html
http://doc.servo.org/style/selector_parser/enum.PseudoElement.html

### LINKED ISSUE #20126: Dead links in style.md
https://github.com/servo/servo/blob/master/docs/components/style.md

These links no longer work:

http://doc.servo.org/style/selector_parser/trait.SelectorImplExt.html
http://doc.servo.org/style/servo_selector_parser/struct.SelectorImpl.html
http://doc.servo.org/selectors/matching/struct.Rule.html
http://doc.servo.org/selectors/matching/struct.SelectorMap.html
http://doc.servo.org/style/stylist/struct.PerPseudoElementSelectorMap.html
http://doc.servo.org/style/stylist/struct.PerOriginSelectorMap.html
http://doc.servo.org/style/selector_parser/enum.PseudoElement.html

### PR DESCRIPTION
**Pull request by sculpt0r**
Replace outdated project name and its link that leads to a webpage with inappropriate content.

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because docs

This PR is related but not entirely fix servo/book#32

### CODE REVIEW

**Comment by atouchet** [comment]
I don't think that Stylo text needs to be changed (aside from the missing space). There's also a more specific Stylo page here https://wiki.mozilla.org/Quantum/Stylo but I don't know if that link is particularly useful as I don't know what was covered in the original page.

**Comment by sculpt0r** [comment]
Well, even webarchive can't tell what was the original content...

In that case I'm not sure what would be the best solution? I would say that the current url should be changed or removed. Also, pointing to any related to the `stylo` page is better than pointing to some weird website.

**Comment by mrobinson** [comment]
> In that case I'm not sure what would be the best solution? I would say that the current url should be changed or removed. Also, pointing to any related to the `stylo` page is better than pointing to some weird website.

Let's just remove the link for now. We need to do a large documentation overhaul and not having the link for now is a good solution in the short-term.

**Comment by sculpt0r** [comment]
@mrobinson - thanks for pointing to the right direction. I've updated the PR, so it could be check again :)

File: docs/components/style.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Boris Zbarsky and Patrick Walton about how style sharing works.
 <a name="selector-impl"></a>
 ## Selector Implementation
 
-In order to be sharable and compatible with [Stylo][stylo](a project that aims 
+In order to be sharable and compatible with `Stylo`(a project that aims
```
Comment (mrobinson) [inline docs/components/style.md:11]:
```suggestion
In order to be sharable and compatible with Stylo (a project that aims 
```

**Review by mrobinson — APPROVED** [review]

**Pull request merged**